pandas/io/parquet.py

```diff
@@ -104,19 +104,23 @@ def write(
             from_pandas_kwargs["preserve_index"] = index
 
         table = self.api.Table.from_pandas(df, **from_pandas_kwargs)
+
+        fs = kwargs.pop("filesystem", get_fs_for_path(path))
```

Comment:
Perhaps we should wait till: https://github.com/pandas-dev/pandas/pull/34266/files#diff-0d7b5a2c72b4dfc11d80afe159d45ff8L153 is merged. Since this method is changing. @TomAugspurger  might have thoughts

Reply:
I agree with that #34266 is more robust solution, which should be v1.1? 

This PR could be a patch for 1.0.x as `to_parquet` with `partition_cols` specified is not working as designed. Without `partition_cols` it works as intended.

Reply:
We original backported this functionality to 1.0.x but it was then reverted in https://github.com/pandas-dev/pandas/pull/34632 -> So there is no s3 directory functionality on 1.0.x currently. See whatsnew: https://pandas.pydata.org/docs/whatsnew/v1.0.5.html#fixed-regressions. So think this will have to wait for 1.1

Reply:
Ok, sounds good. I will alter this after #34266 is merged.